src/common/classes/ConfigHandler.ts

```diff
@@ -73,12 +77,67 @@ export class ConfigHandler {
         this.writeConfig<ITccAutosave>(autosave, filePath, { mode: this.autosaveFileMod });
     }
 
+    private createFanProfile(name: string, rows): ITccFanProfile {
+        return {
+            name: name,
+            tableCPU: rows.map(row => ({"temp": row[0], "speed": row[1]})),
+            tableGPU: rows.map(row => ({"temp": row[0], "speed": row[2]}))
+        }
+    }
+
     readFanTables(filePath: string = this.pathFanTables): ITccFanProfile[] {
-        return this.readConfig<ITccFanProfile[]>(filePath);
+        const lines = fs.readFileSync(filePath, 'utf-8').split('\n');
+        let currentProfile = null;
+        let currentRows = [];
+        let profiles = [];
+        const blankLineRegex = /^\s*(#.*)?$/
+        const sectionLineRegex = /^\s*\[([^\]]+)\]\s*(#.*)?$/
+        const rowLineRegex = /^\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*(#.*)?$/
+        let match
+        for(var line of lines) {
+            if(blankLineRegex.test(line)) {
+                continue;
+            } else if((match = rowLineRegex.exec(line)) !== null) {
+                if(currentProfile === null) {
+                    throw Error(`syntax error reading fan profiles from "${filePath}"`);
+                }
+                currentRows.push([parseInt(match[1]), parseInt(match[2]), parseInt(match[3])]);
+            } else if((match = sectionLineRegex.exec(line)) !== null) {
+                if(currentProfile !== null) {
+                    profiles.push(this.createFanProfile(currentProfile, currentRows));

```

Comment:
What happens if currentRows is null?

Reply:
`currentRows` should never be null but could be an empty list.
are you saying that empty profiles should be discarded? For example
```
[MyProfile]

[MyOtherProfile]
0,0,0
```
only `MyOtherProfile` should be created?